src/main/scala/nl.knaw.dans.easy.bagstore/ItemId.scala

```diff
@@ -33,12 +33,26 @@ abstract class ItemId(val uuid: UUID) {
 }
 
 object ItemId {
+  // FIXME the exact same pattern is also used in dans-bag-lib, share value?
+  val uuidRegex = "[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}"
+
   def fromString(s: String): Try[ItemId] = Try {
     s.split("/", 2) match {
-      case Array(uuidStr) => BagId(UUID.fromString(uuidStr))
+      case Array(uuidStr) => BagId(UUID.fromString(validateUuid(uuidStr.toLowerCase)))
       case Array(uuidStr, path) =>
-        FileId(UUID.fromString(uuidStr), Paths.get(URLDecoder.decode(path, "UTF-8")))
+        FileId(UUID.fromString(validateUuid(uuidStr.toLowerCase())), Paths.get(URLDecoder.decode(path, "UTF-8")))
+    }
+  }
+
+  def validateUuid(uuidAsString: String): String = {
+    if (!(uuidAsString.trim.length == 36)) { //FIXME length of 36 is implicitly checked in pattern below
```

Comment:
Not sure if you keep this check, because of the `FIXME` next to it. But if you keep it, please rewrite this to:

```suggestion
    if (uuidAsString.trim.length != 36) { //FIXME length of 36 is implicitly checked in pattern below
```

I have to see, in regards to the `FIXME`, I probably like it better this way, since you can now provide a more precise error message.